charts/vector/templates/service-headless.yaml

```diff
@@ -1,4 +1,63 @@
-{{- if .Values.service.enabled }}
+{{/*
+TODO: Remove outer "if/elseif"
```

Comment:
Still a valid TODO ?

Reply:
Yeah, we should be able to have just a single `{{- if .Values.serviceHeadless.enabled }}` wrapping the entire resource, just to clean things up. The existing if/elif is needed because the upgrade from the previous version to this version would fail. We can remove it for the upgrade from this version to the next (theoretically, depends on how much grace we want to provide for upgrades).

Reply:
Ah, makes sense.